### PR TITLE
Update hook for pyarrow

### PR DESCRIPTION
**Pull request by n-elie**

### CODE REVIEW

**Comment by rokm** [comment]
This will pull in `pyarrow.tests`, which we definitely do not want:

<details><summary>Example</summary><p>

```
>>> import pprint
>>> pprint.pprint(collect_submodules('pyarrow'))
193 WARNING: Failed to collect submodules for 'pyarrow.tests.parquet' because importing 'pyarrow.tests.parquet' raised: ModuleNotFoundError: No module named 'pytest'
['pyarrow',
 'pyarrow._acero',
 'pyarrow._compute',
 'pyarrow._compute_docstrings',
 'pyarrow._csv',
 'pyarrow._dataset',
 'pyarrow._dataset_orc',
 'pyarrow._dataset_parquet',
 'pyarrow._dataset_parquet_encryption',
 'pyarrow._feather',
 'pyarrow._flight',
 'pyarrow._fs',
 'pyarrow._gcsfs',
 'pyarrow._generated_version',
 'pyarrow._hdfs',
 'pyarrow._hdfsio',
 'pyarrow._json',
 'pyarrow._orc',
 'pyarrow._parquet',
 'pyarrow._parquet_encryption',
 'pyarrow._pyarrow_cpp_tests',
 'pyarrow._s3fs',
 'pyarrow._substrait',
 'pyarrow.acero',
 'pyarrow.benchmark',
 'pyarrow.cffi',
 'pyarrow.compute',
 'pyarrow.conftest',
 'pyarrow.csv',
 'pyarrow.cuda',
 'pyarrow.dataset',
 'pyarrow.feather',
 'pyarrow.filesystem',
 'pyarrow.flight',
 'pyarrow.fs',
 'pyarrow.hdfs',
 'pyarrow.interchange',
 'pyarrow.interchange.buffer',
 'pyarrow.interchange.column',
 'pyarrow.interchange.dataframe',
 'pyarrow.interchange.from_dataframe',
 'pyarrow.ipc',
 'pyarrow.json',
 'pyarrow.jvm',
 'pyarrow.lib',
 'pyarrow.libarrow_python',
 'pyarrow.libarrow_python_flight',
 'pyarrow.orc',
 'pyarrow.pandas_compat',
 'pyarrow.parquet',
 'pyarrow.parquet.core',
 'pyarrow.parquet.encryption',
 'pyarrow.substrait',
 'pyarrow.tests',
 'pyarrow.tests.arrow_16597',
 'pyarrow.tests.arrow_7980',
 'pyarrow.tests.conftest',
 'pyarrow.tests.interchange',
 'pyarrow.tests.interchange.test_conversion',
 'pyarrow.tests.interchange.test_interchange_spec',
 'pyarrow.tests.pandas_examples',
 'pyarrow.tests.pandas_threaded_import',
 'pyarrow.tests.read_record_batch',
 'pyarrow.tests.strategies',
 'pyarrow.tests.test_acero',
 'pyarrow.tests.test_adhoc_memory_leak',
 'pyarrow.tests.test_array',
 'pyarrow.tests.test_builder',
 'pyarrow.tests.test_cffi',
 'pyarrow.tests.test_compute',
 'pyarrow.tests.test_convert_builtin',
 'pyarrow.tests.test_cpp_internals',
 'pyarrow.tests.test_csv',
 'pyarrow.tests.test_cuda',
 'pyarrow.tests.test_cuda_numba_interop',
 'pyarrow.tests.test_cython',
 'pyarrow.tests.test_dataset',
 'pyarrow.tests.test_dataset_encryption',
 'pyarrow.tests.test_deprecations',
 'pyarrow.tests.test_exec_plan',
 'pyarrow.tests.test_extension_type',
 'pyarrow.tests.test_feather',
 'pyarrow.tests.test_filesystem',
 'pyarrow.tests.test_flight',
 'pyarrow.tests.test_flight_async',
 'pyarrow.tests.test_fs',
 'pyarrow.tests.test_gandiva',
 'pyarrow.tests.test_gdb',
 'pyarrow.tests.test_hdfs',
 'pyarrow.tests.test_io',
 'pyarrow.tests.test_ipc',
 'pyarrow.tests.test_json',
 'pyarrow.tests.test_jvm',
 'pyarrow.tests.test_memory',
 'pyarrow.tests.test_misc',
 'pyarrow.tests.test_orc',
 'pyarrow.tests.test_pandas',
 'pyarrow.tests.test_scalars',
 'pyarrow.tests.test_schema',
 'pyarrow.tests.test_sparse_tensor',
 'pyarrow.tests.test_strategies',
 'pyarrow.tests.test_substrait',
 'pyarrow.tests.test_table',
 'pyarrow.tests.test_tensor',
 'pyarrow.tests.test_types',
 'pyarrow.tests.test_udf',
 'pyarrow.tests.test_util',
 'pyarrow.tests.util',
 'pyarrow.types',
 'pyarrow.util',
 'pyarrow.vendored',
 'pyarrow.vendored.docscrape',
 'pyarrow.vendored.version']
```

</p></details>

**Review by rokm — CHANGES_REQUESTED** [review]
At the very least `pyarrow.tests` should be filtered out, e.g.:

```
hiddenimports = collect_submodules('pyarrow', filter=lambda x: not x.startswith('pyarrow.tests'))
```

**Comment by n-elie** [comment]
My bad, I updated the PR according to your suggestions

File: src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-pyarrow.py [hunk 1]
```diff
@@ -12,13 +12,8 @@
...
-from PyInstaller.utils.hooks import collect_data_files, collect_dynamic_libs
-
-hiddenimports = [
-    "pyarrow._parquet",
-    "pyarrow.lib",
-    "pyarrow.compat",
-]
+from PyInstaller.utils.hooks import collect_submodules, collect_data_files, collect_dynamic_libs
 
+hiddenimports = collect_submodules('pyarrow', filter=lambda x: "tests" not in x.split("."))
```
Comment (rokm) [inline src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-pyarrow.py:17]: Splitting the name string is redundant here, isn't it?

Comment (n-elie) [reply src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-pyarrow.py]: You are right, I removed the split

**Comment by rokm** [comment]
And now for the administrative part: please add [a news fragment](https://github.com/pyinstaller/pyinstaller-hooks-contrib/blob/master/news/README.txt) describing the change for the changelog (the `update` category, and something along the lines of "Update hook for ``pyarrow`` to collect all of the package's submodules."). 

Then we should be good to merge.

**Review by rokm — APPROVED** [review]

**Comment by n-elie** [comment]
Thanks for your help

**Pull request merged**